Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -17,18 +17,14 @@ import struct Foundation.URLRequest
 // in the CRT engine so that is why it's a class
 public final class SdkHttpRequest: RequestMessage {
     public var body: ByteStream
-    public let endpoint: Endpoint
+    public var endpoint: Endpoint
     public let method: HttpMethodType
-    private var additionalHeaders: Headers = Headers()
-    public var headers: Headers {
-        var allHeaders = endpoint.headers ?? Headers()
-        allHeaders.addAll(headers: additionalHeaders)
-        return allHeaders
-    }
```

Comment:
Note: this was done as a temporary measure to accommodate a customer request (https://github.com/smithy-lang/smithy-swift/issues/583) for allowing injection of new headers. But now that interceptors are implemented, this can be deleted.